libbeat/processors/uuid/generator/elasticsearch/generator.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elasticsearch
+
+import (
+	"encoding/base64"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var (
+	sequenceNumber uint32
+	lastTimestamp  *time.Time
+	mac            []byte
+	mu             sync.Mutex
+)
+
+func init() {
+	m, err := getSecureMungedMACAddress()
+	if err != nil {
+		panic(err)
+	}
+	mac = m
+	sequenceNumber = rand.Uint32()
+}
+
+// GetBase64UUID returns a base64-encoded, randomly-generated, but roughly ordered (over time), unique
+// ID. The algorithm used to generate the ID is the same as used by Elasticsearch.
+// See https://github.com/elastic/elasticsearch/blob/a666fb2266/server/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java
+func GetBase64UUID() string {
+	mu.Lock()
+	defer mu.Unlock()
+
+	sequenceNumber++
+
+	// We only use bottom 3 bytes for the sequence number.
+	s := sequenceNumber & 0xffffff
```

Comment:
Is there a reason we only use 3 byte, not 4? Although `2^24 = ~16M ids/s` is not too bad, I'd prefer if we could go even higher.

Reply:
No reason other than the fact that this algorithm is currently identical to the one in ES, which also uses only the bottom 3 bytes: 
 https://github.com/elastic/elasticsearch/blob/b8ce07b4cc55fde938a9c88f89b7e6b1e58988e3/server/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java#L60

So it's just a matter of how closely we want to mirror the ES algorithm, I guess.